paymentsheet/src/main/java/com/stripe/android/paymentsheet/SavedPaymentMethodMutator.kt

```diff
@@ -60,17 +61,22 @@ internal class SavedPaymentMethodMutator(
 
     val paymentOptionsItems: StateFlow<List<PaymentOptionsItem>> = paymentOptionsItemsMapper()
 
-    val canEdit: StateFlow<Boolean> = paymentOptionsItems.mapAsStateFlow { items ->
-        val paymentMethods = items.filterIsInstance<PaymentOptionsItem.SavedPaymentMethod>()
-        if (allowsRemovalOfLastSavedPaymentMethod) {
-            paymentMethods.isNotEmpty()
-        } else {
-            if (paymentMethods.size == 1) {
-                // We will allow them to change card brand, but not delete.
-                paymentMethods.first().isModifiable
-            } else {
-                paymentMethods.size > 1
+    val canRemove: StateFlow<Boolean> = customerStateHolder.customer.mapAsStateFlow { customerState ->
+        customerState?.run {
+            when (paymentMethods.size) {
+                0 -> false
+                1 -> allowsRemovalOfLastSavedPaymentMethod
+                else -> true
             }
+        } ?: false
+    }
+
+    val canEdit: StateFlow<Boolean> = combineAsStateFlow(
+        canRemove,
+        paymentOptionsItems
+    ) { canRemove, items ->
+        canRemove || items.filterIsInstance<PaymentOptionsItem.SavedPaymentMethod>().any { item ->
+            item.isModifiable
```

Comment:
Seems like long term, we want to have the isModifiable accessible on a payment method, rather than needing something like `PaymentOptionsItem`.

Reply:
We do have the payment methods available here. We can move the function as an extension of `PaymentMethod` here or in a follow-up.

Reply:
Right, but there's more information that we currently get from our payment options mapper. So we'd need to do something there too. 

I don't think it belongs in this PR. But would be a good follow up.

Reply:
Should be handled by the changes defined [here](https://docs.google.com/document/d/1TnwdgpEBu5QmTspEuOYS2JBRVpt2lJay6FjPP_pC4kk/edit) afterwards.